### ISSUE TITLE
`indent-binary-ops`: Indentation Adjustment When ‘(’ Is Included in the Logical Expression

### ISSUE THREAD

**Issue by k35o** [eslint-stylistic/eslint-stylistic#545]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

There was a formatting behavior related to indentation that changed due to modification A, which caught my attention https://github.com/eslint-stylistic/eslint-stylistic/issues/530, which caught my attention.

after
```ts
const a = (
  (flag1
    && flag2)
    || (flag3
      && flag4)
)
```
before
```ts
const a = (
  (flag1
    && flag2)
  || (flag3
    && flag4)
)
```
Indentation is now deeper when logical expressions are at the same level but contain `(`.

I would like each logical expression to be indented at the same level when using comparison operators., even if it contain `(`.
```ts
// Cases without (
const a = (
  flag1
  || flag2
  || flag3
)

//  Hope it's the same as above.
const b = (
  (flag1
    && flag2)
  || (flag3
    && flag4)
  || (flag5
    && flag6)
)
```

I think this is because `needAdditionIndent` is determined to be `true` in [this section](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/packages/eslint-plugin/rules/indent-binary-ops/indent-binary-ops._plus_.ts#L84).
```
// Before the left token is a opening bracket
(['[', '(', '=>', ':'].includes(tokenBeforeAll?.value || '') && firstTokenOfLineLeft?.loc.start.line === tokenBeforeAll?.loc.start.line)
```

### Reproduction

https://stackblitz.com/edit/github-d1geg2?file=src%2Findex.ts

### Contributes

- [X] I am willing to submit a PR to fix this issue
- [X] I am willing to submit a PR with failing tests

**Comment by Anoesj** [comment]
Not entirely sure this is the same or a similar scenario, but this one caught my attention when trying to configure Stylistic to taste:
![image](https://github.com/user-attachments/assets/1a9f0634-b5c0-44f0-8ca1-5ef95d4355e6)

I use the following settings for `indent`:
```
'@stylistic/indent': ['error', 2, {
  VariableDeclarator: 'first',
  SwitchCase: 1,
}],
```

This results in the following error in the scenario above: `Expected indentation of 4 spaces eslint[@stylistic/indent-binary-ops](https://eslint.style/rules/plus/indent-binary-ops)`

It wants to "correct" it to:
```ts
const hasZero = (
  firstCheck
  && [0, 1, 2]
    .includes(0)
    && secondCheck
);
```

This looks very weird to me, as `&& secondCheck` instinctively should be at the same level as the first two checks.

It doesn't seem possible to configure `@stylistic/indent-binary-ops` to accept this, so I've turned it off entirely for now 😐
